apps/prairielearn/src/admin_queries/generate_submissions.ts

```diff
@@ -0,0 +1,147 @@
+import { mapSeries } from 'async';
+import { z } from 'zod';
+
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { config } from '../lib/config.js';
+import {
+  CourseSchema,
+  InstanceQuestionSchema,
+  QuestionSchema,
+  UserSchema,
+} from '../lib/db-types.js';
+import { saveSubmission } from '../lib/grading.js';
+import { TEST_TYPES, type TestType, createTestSubmissionData } from '../lib/question-testing.js';
+import { ensureVariant } from '../lib/question-variant.js';
+import { selectOptionalAssessmentById } from '../models/assessment.js';
+import { selectOptionalCourseInstanceById } from '../models/course-instances.js';
+import { selectCourseById } from '../models/course.js';
+
+import { type AdministratorQueryResult, type AdministratorQuerySpecs } from './lib/util.js';
+
+export const specs: AdministratorQuerySpecs = {
+  description:
+    'Creates a random submission for each assessment instance in an assessment. Assessment instances must already exist and be open. Submission is saved, but not graded.',
+  enabled: config.devMode, // This query is dangerous in production environments, so it is only enabled in dev mode
```

Comment:
🙏 thanks for this. This will work, though we may want to make this a function in the future to delay the evaluation until the config has actually been loaded. Currently this always comes from `NODE_ENV` which is known at process start, so that's not an issue for now.

Reply:
Admin queries are loaded dynamically, so config should be loaded by the time this is computed.